### ISSUE TITLE
[Feature]: Option to use artifacts but disable syncing

### ISSUE THREAD

**Issue by hyerra** [wandb/wandb#5712]
### Description

I want to be able to pull artifacts like pretrained models, datasets, etc. without necessarily uploading the results of my run to WandB. This is useful when I'm running small experiments and don't want to necessarily publish the results and clog up the dashboard with a lot of runs.

### Suggested Solution

An extra command in the CLI like "readonly" or something similar which indicates that artifacts and other necessary items are downloaded but nothing is published. It would also be a nice feature to have an option to do this in the Python Library as well.

### Alternatives

I've tried using anonymous="must", but for some reason my results are still logged to the dashboard. Maybe I have to logout? However, it would be much more simpler if we could just have another command in the CLI to prevent logging results.

**Comment by hyerra** [comment]
Is this the same thing as running `wandb offline` in the terminal? I just tried that, but the only problem is I can't use artifacts. I get this error:
`    raise TypeError("Cannot use artifact when in offline mode.")
TypeError: Cannot use artifact when in offline mode.`

I ideally want to still be able to use artifacts and other things stored in the cloud, but just not write new information.

**Comment by hyerra** [comment]
Yep exactly @thanos-wandb. Essentially, the code depends on artifacts from wandb, so I need to pull these. However, since it’s just a local test, I don’t want it to create a new run or log anything to wandb.

**Comment by hyerra** [comment]
Sounds good! Also, would it be possible to add support to still pull these files in offline mode from the cloud if they do not exist locally or if there are new versions?

Like for example let's say we have version 8 of a model but version 9 is there on the cloud. It should prefer using version 9 and download this version. Maybe in this case we could call it readonly mode or something rather than offline mode.

**Comment by thanos-wandb** [comment]
Sure thing! I have just updated the internal ticket to include this comment too. Thank you @hyerra!
